packages/bot/src/bot.ts

```diff
@@ -0,0 +1,1064 @@
+
+export type UserData = {
+    /** The user ID of the user */
+    userId: string
+    /** The username of the user */
+    username: string | null
+    /** The display name of the user */
+    displayName: string | null
+    /** The ENS address of the user */
+    ensAddress?: string
+    /** The bio of the user */
+    bio: string | null
+    /** The NFT that the user is currently showcasing */
+    nft?: {
+        tokenId: string
+        contractAddress: string
+        chainId: number
+    }
+    /** URL that points to the profile picture of the user */
+    profilePictureUrl: string
+}
+export type BotEvents = {
+    message: (
+        handler: BotActions,
+        event: BasePayload & {
+            /** The decrypted message content */
+            message: string
+            /** You can use this to check if the message is a direct message */
+            isDm: boolean
+            /** You can use this to check if the message is a group message */
+            isGdm: boolean
+        },
+    ) => void | Promise<void>
+    redaction: (
+        handler: BotActions,
+        event: BasePayload & {
+            /** The event ID that got redacted */
+            refEventId: string
+        },
+    ) => void | Promise<void>
+    messageEdit: (
+        handler: BotActions,
+        event: BasePayload & {
+            /** The event ID of the message that got edited */
+            refEventId: string
+            /** New message */
+            message: string
+        },
+    ) => void | Promise<void>
+    mentioned: (
+        handler: BotActions,
+        event: BasePayload & {
+            /** The decrypted message content */
+            message: string
+        },
+    ) => void | Promise<void>
+    reply: (
+        handler: BotActions,
+        event: BasePayload & {
+            /** The decrypted message content */
+            message: string
+        },
+    ) => void | Promise<void>
+    reaction: (
+        handler: BotActions,
+        event: BasePayload & {
+            /** The reaction that was added */
+            reaction: string
+            /** The event ID of the message that got reacted to */
+            messageId: string
+            /** The user ID of the user that added the reaction */
+            userId: string
+        },
+    ) => Promise<void> | void
+    eventRevoke: (
+        handler: BotActions,
+        event: BasePayload & {
+            /** The event ID of the message that got revoked */
+            refEventId: string
+        },
+    ) => Promise<void> | void
+    // TODO:
+    tip: (
+        handler: BotActions,
+        event: BasePayload & {
+            /** The amount of the tip */
+            amount: bigint
+            /** The currency of the tip */
+            currency: `0x${string}`
+        },
+    ) => Promise<void> | void
+    channelJoin: (handler: BotActions, event: BasePayload) => Promise<void> | void
+    channelLeave: (handler: BotActions, event: BasePayload) => Promise<void> | void
```

Comment:
I need to add a spaceJoin and spaceLeave and add the `setDisplayName` and `setUsername` there